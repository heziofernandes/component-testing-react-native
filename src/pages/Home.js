import React from "react";
import { StyleSheet, Text } from "react-native";

export default () => {
  return (
    <Text 
      testID="Home.welcomeMsg"
      style={styles.headerText}>
        Welcome Back!
    </Text>
      );
  };

  const styles = StyleSheet.create({
    headerText: {
    color: "#353031",
    fontWeight: "bold",
    fontSize: 20,
    marginBottom: 10,
    textAlign:"center"
    },
  });