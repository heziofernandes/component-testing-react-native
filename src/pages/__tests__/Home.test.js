import React from "react";
import { render } from "@testing-library/react-native";

import Home from "../Home";

it("Valid welcome mensage", async () => {
  const { getByText, queryAllByText } = render(<Home />);
  
  getByText("Welcome Back!");
  expect(queryAllByText("Welcome Back!").length).toBe(1);
  expect(queryAllByText("Welcome Back!")).toMatchSnapshot();
});