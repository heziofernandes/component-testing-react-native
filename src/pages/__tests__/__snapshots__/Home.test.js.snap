// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Valid welcome mensage 1`] = `
Array [
  ReactTestInstance {
    "_fiber": FiberNode {
      "_debugHookTypes": null,
      "_debugID": 4,
      "_debugIsCurrentlyTiming": false,
      "_debugNeedsRemount": false,
      "_debugOwner": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 3,
        "_debugIsCurrentlyTiming": false,
        "_debugNeedsRemount": false,
        "_debugOwner": null,
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": null,
        "child": [Circular],
        "childExpirationTime": 0,
        "dependencies": null,
        "effectTag": 1,
        "elementType": [Function],
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": Object {},
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": Object {},
        "ref": null,
        "return": FiberNode {
          "_debugHookTypes": null,
          "_debugID": 1,
          "_debugIsCurrentlyTiming": false,
          "_debugNeedsRemount": false,
          "_debugOwner": null,
          "_debugSource": null,
          "actualDuration": 0,
          "actualStartTime": -1,
          "alternate": FiberNode {
            "_debugHookTypes": null,
            "_debugID": 1,
            "_debugIsCurrentlyTiming": false,
            "_debugNeedsRemount": false,
            "_debugOwner": null,
            "_debugSource": null,
            "actualDuration": 0,
            "actualStartTime": -1,
            "alternate": [Circular],
            "child": null,
            "childExpirationTime": 0,
            "dependencies": null,
            "effectTag": 0,
            "elementType": null,
            "expirationTime": 1073741823,
            "firstEffect": null,
            "index": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": null,
            "memoizedState": null,
            "mode": 0,
            "nextEffect": null,
            "pendingProps": null,
            "ref": null,
            "return": null,
            "selfBaseDuration": 0,
            "sibling": null,
            "stateNode": FiberRootNode {
              "callbackExpirationTime": 0,
              "callbackNode": null,
              "callbackPriority": 90,
              "containerInfo": Object {
                "children": Array [
                  Object {
                    "children": Array [
                      Object {
                        "isHidden": false,
                        "tag": "TEXT",
                        "text": "Welcome Back!",
                      },
                    ],
                    "internalInstanceHandle": FiberNode {
                      "_debugHookTypes": null,
                      "_debugID": 5,
                      "_debugIsCurrentlyTiming": false,
                      "_debugNeedsRemount": false,
                      "_debugOwner": [Circular],
                      "_debugSource": null,
                      "actualDuration": 0,
                      "actualStartTime": -1,
                      "alternate": null,
                      "child": FiberNode {
                        "_debugHookTypes": null,
                        "_debugID": 6,
                        "_debugIsCurrentlyTiming": false,
                        "_debugNeedsRemount": false,
                        "_debugOwner": null,
                        "_debugSource": null,
                        "actualDuration": 0,
                        "actualStartTime": -1,
                        "alternate": null,
                        "child": null,
                        "childExpirationTime": 0,
                        "dependencies": null,
                        "effectTag": 0,
                        "elementType": null,
                        "expirationTime": 0,
                        "firstEffect": null,
                        "index": 0,
                        "key": null,
                        "lastEffect": null,
                        "memoizedProps": "Welcome Back!",
                        "memoizedState": null,
                        "mode": 0,
                        "nextEffect": null,
                        "pendingProps": "Welcome Back!",
                        "ref": null,
                        "return": [Circular],
                        "selfBaseDuration": 0,
                        "sibling": null,
                        "stateNode": Object {
                          "isHidden": false,
                          "tag": "TEXT",
                          "text": "Welcome Back!",
                        },
                        "tag": 6,
                        "treeBaseDuration": 0,
                        "type": null,
                        "updateQueue": null,
                      },
                      "childExpirationTime": 0,
                      "dependencies": null,
                      "effectTag": 0,
                      "elementType": "Text",
                      "expirationTime": 0,
                      "firstEffect": null,
                      "index": 0,
                      "key": null,
                      "lastEffect": null,
                      "memoizedProps": Object {
                        "children": "Welcome Back!",
                        "style": Object {
                          "color": "#353031",
                          "fontSize": 20,
                          "fontWeight": "bold",
                          "marginBottom": 10,
                          "textAlign": "center",
                        },
                        "testID": "Home.welcomeMsg",
                      },
                      "memoizedState": null,
                      "mode": 0,
                      "nextEffect": null,
                      "pendingProps": Object {
                        "children": "Welcome Back!",
                        "style": Object {
                          "color": "#353031",
                          "fontSize": 20,
                          "fontWeight": "bold",
                          "marginBottom": 10,
                          "textAlign": "center",
                        },
                        "testID": "Home.welcomeMsg",
                      },
                      "ref": null,
                      "return": [Circular],
                      "selfBaseDuration": 0,
                      "sibling": null,
                      "stateNode": [Circular],
                      "tag": 5,
                      "treeBaseDuration": 0,
                      "type": "Text",
                      "updateQueue": null,
                    },
                    "isHidden": false,
                    "props": Object {
                      "children": "Welcome Back!",
                      "style": Object {
                        "color": "#353031",
                        "fontSize": 20,
                        "fontWeight": "bold",
                        "marginBottom": 10,
                        "textAlign": "center",
                      },
                      "testID": "Home.welcomeMsg",
                    },
                    "rootContainerInstance": [Circular],
                    "tag": "INSTANCE",
                    "type": "Text",
                  },
                ],
                "createNodeMock": [Function],
                "tag": "CONTAINER",
              },
              "context": Object {},
              "current": [Circular],
              "finishedExpirationTime": 0,
              "finishedWork": null,
              "firstPendingTime": 0,
              "firstSuspendedTime": 0,
              "hydrate": false,
              "interactionThreadID": 1,
              "lastExpiredTime": 0,
              "lastPingedTime": 0,
              "lastSuspendedTime": 0,
              "memoizedInteractions": Set {},
              "nextKnownPendingLevel": 0,
              "pendingChildren": null,
              "pendingContext": null,
              "pendingInteractionMap": Map {},
              "pingCache": null,
              "tag": 0,
              "timeoutHandle": -1,
            },
            "tag": 3,
            "treeBaseDuration": 0,
            "type": null,
            "updateQueue": Object {
              "baseState": null,
              "firstCapturedEffect": null,
              "firstCapturedUpdate": null,
              "firstEffect": null,
              "firstUpdate": Object {
                "callback": null,
                "expirationTime": 1073741823,
                "next": null,
                "nextEffect": null,
                "payload": Object {
                  "element": <_default />,
                },
                "priority": 97,
                "suspenseConfig": null,
                "tag": 0,
              },
              "lastCapturedEffect": null,
              "lastCapturedUpdate": null,
              "lastEffect": null,
              "lastUpdate": Object {
                "callback": null,
                "expirationTime": 1073741823,
                "next": null,
                "nextEffect": null,
                "payload": Object {
                  "element": <_default />,
                },
                "priority": 97,
                "suspenseConfig": null,
                "tag": 0,
              },
            },
          },
          "child": [Circular],
          "childExpirationTime": 0,
          "dependencies": null,
          "effectTag": 0,
          "elementType": null,
          "expirationTime": 0,
          "firstEffect": [Circular],
          "index": 0,
          "key": null,
          "lastEffect": [Circular],
          "memoizedProps": null,
          "memoizedState": Object {
            "element": <_default />,
          },
          "mode": 0,
          "nextEffect": null,
          "pendingProps": null,
          "ref": null,
          "return": null,
          "selfBaseDuration": 0,
          "sibling": null,
          "stateNode": FiberRootNode {
            "callbackExpirationTime": 0,
            "callbackNode": null,
            "callbackPriority": 90,
            "containerInfo": Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "isHidden": false,
                      "tag": "TEXT",
                      "text": "Welcome Back!",
                    },
                  ],
                  "internalInstanceHandle": FiberNode {
                    "_debugHookTypes": null,
                    "_debugID": 5,
                    "_debugIsCurrentlyTiming": false,
                    "_debugNeedsRemount": false,
                    "_debugOwner": [Circular],
                    "_debugSource": null,
                    "actualDuration": 0,
                    "actualStartTime": -1,
                    "alternate": null,
                    "child": FiberNode {
                      "_debugHookTypes": null,
                      "_debugID": 6,
                      "_debugIsCurrentlyTiming": false,
                      "_debugNeedsRemount": false,
                      "_debugOwner": null,
                      "_debugSource": null,
                      "actualDuration": 0,
                      "actualStartTime": -1,
                      "alternate": null,
                      "child": null,
                      "childExpirationTime": 0,
                      "dependencies": null,
                      "effectTag": 0,
                      "elementType": null,
                      "expirationTime": 0,
                      "firstEffect": null,
                      "index": 0,
                      "key": null,
                      "lastEffect": null,
                      "memoizedProps": "Welcome Back!",
                      "memoizedState": null,
                      "mode": 0,
                      "nextEffect": null,
                      "pendingProps": "Welcome Back!",
                      "ref": null,
                      "return": [Circular],
                      "selfBaseDuration": 0,
                      "sibling": null,
                      "stateNode": Object {
                        "isHidden": false,
                        "tag": "TEXT",
                        "text": "Welcome Back!",
                      },
                      "tag": 6,
                      "treeBaseDuration": 0,
                      "type": null,
                      "updateQueue": null,
                    },
                    "childExpirationTime": 0,
                    "dependencies": null,
                    "effectTag": 0,
                    "elementType": "Text",
                    "expirationTime": 0,
                    "firstEffect": null,
                    "index": 0,
                    "key": null,
                    "lastEffect": null,
                    "memoizedProps": Object {
                      "children": "Welcome Back!",
                      "style": Object {
                        "color": "#353031",
                        "fontSize": 20,
                        "fontWeight": "bold",
                        "marginBottom": 10,
                        "textAlign": "center",
                      },
                      "testID": "Home.welcomeMsg",
                    },
                    "memoizedState": null,
                    "mode": 0,
                    "nextEffect": null,
                    "pendingProps": Object {
                      "children": "Welcome Back!",
                      "style": Object {
                        "color": "#353031",
                        "fontSize": 20,
                        "fontWeight": "bold",
                        "marginBottom": 10,
                        "textAlign": "center",
                      },
                      "testID": "Home.welcomeMsg",
                    },
                    "ref": null,
                    "return": [Circular],
                    "selfBaseDuration": 0,
                    "sibling": null,
                    "stateNode": [Circular],
                    "tag": 5,
                    "treeBaseDuration": 0,
                    "type": "Text",
                    "updateQueue": null,
                  },
                  "isHidden": false,
                  "props": Object {
                    "children": "Welcome Back!",
                    "style": Object {
                      "color": "#353031",
                      "fontSize": 20,
                      "fontWeight": "bold",
                      "marginBottom": 10,
                      "textAlign": "center",
                    },
                    "testID": "Home.welcomeMsg",
                  },
                  "rootContainerInstance": [Circular],
                  "tag": "INSTANCE",
                  "type": "Text",
                },
              ],
              "createNodeMock": [Function],
              "tag": "CONTAINER",
            },
            "context": Object {},
            "current": [Circular],
            "finishedExpirationTime": 0,
            "finishedWork": null,
            "firstPendingTime": 0,
            "firstSuspendedTime": 0,
            "hydrate": false,
            "interactionThreadID": 1,
            "lastExpiredTime": 0,
            "lastPingedTime": 0,
            "lastSuspendedTime": 0,
            "memoizedInteractions": Set {},
            "nextKnownPendingLevel": 0,
            "pendingChildren": null,
            "pendingContext": null,
            "pendingInteractionMap": Map {},
            "pingCache": null,
            "tag": 0,
            "timeoutHandle": -1,
          },
          "tag": 3,
          "treeBaseDuration": 0,
          "type": null,
          "updateQueue": Object {
            "baseState": Object {
              "element": <_default />,
            },
            "firstCapturedEffect": null,
            "firstCapturedUpdate": null,
            "firstEffect": null,
            "firstUpdate": null,
            "lastCapturedEffect": null,
            "lastCapturedUpdate": null,
            "lastEffect": null,
            "lastUpdate": null,
          },
        },
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": null,
        "tag": 0,
        "treeBaseDuration": 0,
        "type": [Function],
        "updateQueue": null,
      },
      "_debugSource": null,
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": null,
      "child": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 5,
        "_debugIsCurrentlyTiming": false,
        "_debugNeedsRemount": false,
        "_debugOwner": [Circular],
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": null,
        "child": FiberNode {
          "_debugHookTypes": null,
          "_debugID": 6,
          "_debugIsCurrentlyTiming": false,
          "_debugNeedsRemount": false,
          "_debugOwner": null,
          "_debugSource": null,
          "actualDuration": 0,
          "actualStartTime": -1,
          "alternate": null,
          "child": null,
          "childExpirationTime": 0,
          "dependencies": null,
          "effectTag": 0,
          "elementType": null,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": "Welcome Back!",
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": "Welcome Back!",
          "ref": null,
          "return": [Circular],
          "selfBaseDuration": 0,
          "sibling": null,
          "stateNode": Object {
            "isHidden": false,
            "tag": "TEXT",
            "text": "Welcome Back!",
          },
          "tag": 6,
          "treeBaseDuration": 0,
          "type": null,
          "updateQueue": null,
        },
        "childExpirationTime": 0,
        "dependencies": null,
        "effectTag": 0,
        "elementType": "Text",
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": Object {
          "children": "Welcome Back!",
          "style": Object {
            "color": "#353031",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 10,
            "textAlign": "center",
          },
          "testID": "Home.welcomeMsg",
        },
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": Object {
          "children": "Welcome Back!",
          "style": Object {
            "color": "#353031",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 10,
            "textAlign": "center",
          },
          "testID": "Home.welcomeMsg",
        },
        "ref": null,
        "return": [Circular],
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": Object {
          "children": Array [
            Object {
              "isHidden": false,
              "tag": "TEXT",
              "text": "Welcome Back!",
            },
          ],
          "internalInstanceHandle": [Circular],
          "isHidden": false,
          "props": Object {
            "children": "Welcome Back!",
            "style": Object {
              "color": "#353031",
              "fontSize": 20,
              "fontWeight": "bold",
              "marginBottom": 10,
              "textAlign": "center",
            },
            "testID": "Home.welcomeMsg",
          },
          "rootContainerInstance": Object {
            "children": Array [
              [Circular],
            ],
            "createNodeMock": [Function],
            "tag": "CONTAINER",
          },
          "tag": "INSTANCE",
          "type": "Text",
        },
        "tag": 5,
        "treeBaseDuration": 0,
        "type": "Text",
        "updateQueue": null,
      },
      "childExpirationTime": 0,
      "dependencies": null,
      "effectTag": 1,
      "elementType": [Function],
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "children": "Welcome Back!",
        "style": Object {
          "color": "#353031",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 10,
          "textAlign": "center",
        },
        "testID": "Home.welcomeMsg",
      },
      "memoizedState": null,
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "children": "Welcome Back!",
        "style": Object {
          "color": "#353031",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 10,
          "textAlign": "center",
        },
        "testID": "Home.welcomeMsg",
      },
      "ref": null,
      "return": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 3,
        "_debugIsCurrentlyTiming": false,
        "_debugNeedsRemount": false,
        "_debugOwner": null,
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": null,
        "child": [Circular],
        "childExpirationTime": 0,
        "dependencies": null,
        "effectTag": 1,
        "elementType": [Function],
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": Object {},
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": Object {},
        "ref": null,
        "return": FiberNode {
          "_debugHookTypes": null,
          "_debugID": 1,
          "_debugIsCurrentlyTiming": false,
          "_debugNeedsRemount": false,
          "_debugOwner": null,
          "_debugSource": null,
          "actualDuration": 0,
          "actualStartTime": -1,
          "alternate": FiberNode {
            "_debugHookTypes": null,
            "_debugID": 1,
            "_debugIsCurrentlyTiming": false,
            "_debugNeedsRemount": false,
            "_debugOwner": null,
            "_debugSource": null,
            "actualDuration": 0,
            "actualStartTime": -1,
            "alternate": [Circular],
            "child": null,
            "childExpirationTime": 0,
            "dependencies": null,
            "effectTag": 0,
            "elementType": null,
            "expirationTime": 1073741823,
            "firstEffect": null,
            "index": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": null,
            "memoizedState": null,
            "mode": 0,
            "nextEffect": null,
            "pendingProps": null,
            "ref": null,
            "return": null,
            "selfBaseDuration": 0,
            "sibling": null,
            "stateNode": FiberRootNode {
              "callbackExpirationTime": 0,
              "callbackNode": null,
              "callbackPriority": 90,
              "containerInfo": Object {
                "children": Array [
                  Object {
                    "children": Array [
                      Object {
                        "isHidden": false,
                        "tag": "TEXT",
                        "text": "Welcome Back!",
                      },
                    ],
                    "internalInstanceHandle": FiberNode {
                      "_debugHookTypes": null,
                      "_debugID": 5,
                      "_debugIsCurrentlyTiming": false,
                      "_debugNeedsRemount": false,
                      "_debugOwner": [Circular],
                      "_debugSource": null,
                      "actualDuration": 0,
                      "actualStartTime": -1,
                      "alternate": null,
                      "child": FiberNode {
                        "_debugHookTypes": null,
                        "_debugID": 6,
                        "_debugIsCurrentlyTiming": false,
                        "_debugNeedsRemount": false,
                        "_debugOwner": null,
                        "_debugSource": null,
                        "actualDuration": 0,
                        "actualStartTime": -1,
                        "alternate": null,
                        "child": null,
                        "childExpirationTime": 0,
                        "dependencies": null,
                        "effectTag": 0,
                        "elementType": null,
                        "expirationTime": 0,
                        "firstEffect": null,
                        "index": 0,
                        "key": null,
                        "lastEffect": null,
                        "memoizedProps": "Welcome Back!",
                        "memoizedState": null,
                        "mode": 0,
                        "nextEffect": null,
                        "pendingProps": "Welcome Back!",
                        "ref": null,
                        "return": [Circular],
                        "selfBaseDuration": 0,
                        "sibling": null,
                        "stateNode": Object {
                          "isHidden": false,
                          "tag": "TEXT",
                          "text": "Welcome Back!",
                        },
                        "tag": 6,
                        "treeBaseDuration": 0,
                        "type": null,
                        "updateQueue": null,
                      },
                      "childExpirationTime": 0,
                      "dependencies": null,
                      "effectTag": 0,
                      "elementType": "Text",
                      "expirationTime": 0,
                      "firstEffect": null,
                      "index": 0,
                      "key": null,
                      "lastEffect": null,
                      "memoizedProps": Object {
                        "children": "Welcome Back!",
                        "style": Object {
                          "color": "#353031",
                          "fontSize": 20,
                          "fontWeight": "bold",
                          "marginBottom": 10,
                          "textAlign": "center",
                        },
                        "testID": "Home.welcomeMsg",
                      },
                      "memoizedState": null,
                      "mode": 0,
                      "nextEffect": null,
                      "pendingProps": Object {
                        "children": "Welcome Back!",
                        "style": Object {
                          "color": "#353031",
                          "fontSize": 20,
                          "fontWeight": "bold",
                          "marginBottom": 10,
                          "textAlign": "center",
                        },
                        "testID": "Home.welcomeMsg",
                      },
                      "ref": null,
                      "return": [Circular],
                      "selfBaseDuration": 0,
                      "sibling": null,
                      "stateNode": [Circular],
                      "tag": 5,
                      "treeBaseDuration": 0,
                      "type": "Text",
                      "updateQueue": null,
                    },
                    "isHidden": false,
                    "props": Object {
                      "children": "Welcome Back!",
                      "style": Object {
                        "color": "#353031",
                        "fontSize": 20,
                        "fontWeight": "bold",
                        "marginBottom": 10,
                        "textAlign": "center",
                      },
                      "testID": "Home.welcomeMsg",
                    },
                    "rootContainerInstance": [Circular],
                    "tag": "INSTANCE",
                    "type": "Text",
                  },
                ],
                "createNodeMock": [Function],
                "tag": "CONTAINER",
              },
              "context": Object {},
              "current": [Circular],
              "finishedExpirationTime": 0,
              "finishedWork": null,
              "firstPendingTime": 0,
              "firstSuspendedTime": 0,
              "hydrate": false,
              "interactionThreadID": 1,
              "lastExpiredTime": 0,
              "lastPingedTime": 0,
              "lastSuspendedTime": 0,
              "memoizedInteractions": Set {},
              "nextKnownPendingLevel": 0,
              "pendingChildren": null,
              "pendingContext": null,
              "pendingInteractionMap": Map {},
              "pingCache": null,
              "tag": 0,
              "timeoutHandle": -1,
            },
            "tag": 3,
            "treeBaseDuration": 0,
            "type": null,
            "updateQueue": Object {
              "baseState": null,
              "firstCapturedEffect": null,
              "firstCapturedUpdate": null,
              "firstEffect": null,
              "firstUpdate": Object {
                "callback": null,
                "expirationTime": 1073741823,
                "next": null,
                "nextEffect": null,
                "payload": Object {
                  "element": <_default />,
                },
                "priority": 97,
                "suspenseConfig": null,
                "tag": 0,
              },
              "lastCapturedEffect": null,
              "lastCapturedUpdate": null,
              "lastEffect": null,
              "lastUpdate": Object {
                "callback": null,
                "expirationTime": 1073741823,
                "next": null,
                "nextEffect": null,
                "payload": Object {
                  "element": <_default />,
                },
                "priority": 97,
                "suspenseConfig": null,
                "tag": 0,
              },
            },
          },
          "child": [Circular],
          "childExpirationTime": 0,
          "dependencies": null,
          "effectTag": 0,
          "elementType": null,
          "expirationTime": 0,
          "firstEffect": [Circular],
          "index": 0,
          "key": null,
          "lastEffect": [Circular],
          "memoizedProps": null,
          "memoizedState": Object {
            "element": <_default />,
          },
          "mode": 0,
          "nextEffect": null,
          "pendingProps": null,
          "ref": null,
          "return": null,
          "selfBaseDuration": 0,
          "sibling": null,
          "stateNode": FiberRootNode {
            "callbackExpirationTime": 0,
            "callbackNode": null,
            "callbackPriority": 90,
            "containerInfo": Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "isHidden": false,
                      "tag": "TEXT",
                      "text": "Welcome Back!",
                    },
                  ],
                  "internalInstanceHandle": FiberNode {
                    "_debugHookTypes": null,
                    "_debugID": 5,
                    "_debugIsCurrentlyTiming": false,
                    "_debugNeedsRemount": false,
                    "_debugOwner": [Circular],
                    "_debugSource": null,
                    "actualDuration": 0,
                    "actualStartTime": -1,
                    "alternate": null,
                    "child": FiberNode {
                      "_debugHookTypes": null,
                      "_debugID": 6,
                      "_debugIsCurrentlyTiming": false,
                      "_debugNeedsRemount": false,
                      "_debugOwner": null,
                      "_debugSource": null,
                      "actualDuration": 0,
                      "actualStartTime": -1,
                      "alternate": null,
                      "child": null,
                      "childExpirationTime": 0,
                      "dependencies": null,
                      "effectTag": 0,
                      "elementType": null,
                      "expirationTime": 0,
                      "firstEffect": null,
                      "index": 0,
                      "key": null,
                      "lastEffect": null,
                      "memoizedProps": "Welcome Back!",
                      "memoizedState": null,
                      "mode": 0,
                      "nextEffect": null,
                      "pendingProps": "Welcome Back!",
                      "ref": null,
                      "return": [Circular],
                      "selfBaseDuration": 0,
                      "sibling": null,
                      "stateNode": Object {
                        "isHidden": false,
                        "tag": "TEXT",
                        "text": "Welcome Back!",
                      },
                      "tag": 6,
                      "treeBaseDuration": 0,
                      "type": null,
                      "updateQueue": null,
                    },
                    "childExpirationTime": 0,
                    "dependencies": null,
                    "effectTag": 0,
                    "elementType": "Text",
                    "expirationTime": 0,
                    "firstEffect": null,
                    "index": 0,
                    "key": null,
                    "lastEffect": null,
                    "memoizedProps": Object {
                      "children": "Welcome Back!",
                      "style": Object {
                        "color": "#353031",
                        "fontSize": 20,
                        "fontWeight": "bold",
                        "marginBottom": 10,
                        "textAlign": "center",
                      },
                      "testID": "Home.welcomeMsg",
                    },
                    "memoizedState": null,
                    "mode": 0,
                    "nextEffect": null,
                    "pendingProps": Object {
                      "children": "Welcome Back!",
                      "style": Object {
                        "color": "#353031",
                        "fontSize": 20,
                        "fontWeight": "bold",
                        "marginBottom": 10,
                        "textAlign": "center",
                      },
                      "testID": "Home.welcomeMsg",
                    },
                    "ref": null,
                    "return": [Circular],
                    "selfBaseDuration": 0,
                    "sibling": null,
                    "stateNode": [Circular],
                    "tag": 5,
                    "treeBaseDuration": 0,
                    "type": "Text",
                    "updateQueue": null,
                  },
                  "isHidden": false,
                  "props": Object {
                    "children": "Welcome Back!",
                    "style": Object {
                      "color": "#353031",
                      "fontSize": 20,
                      "fontWeight": "bold",
                      "marginBottom": 10,
                      "textAlign": "center",
                    },
                    "testID": "Home.welcomeMsg",
                  },
                  "rootContainerInstance": [Circular],
                  "tag": "INSTANCE",
                  "type": "Text",
                },
              ],
              "createNodeMock": [Function],
              "tag": "CONTAINER",
            },
            "context": Object {},
            "current": [Circular],
            "finishedExpirationTime": 0,
            "finishedWork": null,
            "firstPendingTime": 0,
            "firstSuspendedTime": 0,
            "hydrate": false,
            "interactionThreadID": 1,
            "lastExpiredTime": 0,
            "lastPingedTime": 0,
            "lastSuspendedTime": 0,
            "memoizedInteractions": Set {},
            "nextKnownPendingLevel": 0,
            "pendingChildren": null,
            "pendingContext": null,
            "pendingInteractionMap": Map {},
            "pingCache": null,
            "tag": 0,
            "timeoutHandle": -1,
          },
          "tag": 3,
          "treeBaseDuration": 0,
          "type": null,
          "updateQueue": Object {
            "baseState": Object {
              "element": <_default />,
            },
            "firstCapturedEffect": null,
            "firstCapturedUpdate": null,
            "firstEffect": null,
            "firstUpdate": null,
            "lastCapturedEffect": null,
            "lastCapturedUpdate": null,
            "lastEffect": null,
            "lastUpdate": null,
          },
        },
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": null,
        "tag": 0,
        "treeBaseDuration": 0,
        "type": [Function],
        "updateQueue": null,
      },
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": Component {
        "_reactInternalFiber": [Circular],
        "_reactInternalInstance": Object {},
        "context": Object {},
        "props": Object {
          "children": "Welcome Back!",
          "style": Object {
            "color": "#353031",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 10,
            "textAlign": "center",
          },
          "testID": "Home.welcomeMsg",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
        },
      },
      "tag": 1,
      "treeBaseDuration": 0,
      "type": [Function],
      "updateQueue": null,
    },
  },
]
`;
